rpc/rpc.h

```diff
@@ -168,9 +168,16 @@ namespace rpc
         // the default allocator is defined in iovector.h/cpp
         virtual void set_allocator(IOAlloc allocation) = 0;
 
-        virtual int shutdown_no_wait() = 0;
+        /**
+         * @brief Shutdown the rpc server from outside.
+         * @param passive Will wait until all clients close the connections
+         * @warning DO NOT invoke this function within the RPC request.
+         *          You should create a thread to invoke it, or just use shutdown_inplace.
+         */
+        virtual int shutdown(bool passive = false) = 0;
 
-        virtual int shutdown() = 0;
+        /** @brief Shutdown the server within a RPC request */
+        virtual void shutdown_inplace(bool passive = false) = 0;
```

Comment:
shutdown_inplace() seems not necessary, because all handlers are async by themselves now.